silhouette/app/com/mohiva/play/silhouette/api/services/AuthenticatorService.scala

```diff
@@ -128,7 +150,7 @@ trait AuthenticatorService[T <: Authenticator] {
    * @param authenticator The authenticator to touch.
    * @return The touched authenticator on the left or the untouched authenticator on the right.
    */
-  protected[silhouette] def touch(authenticator: T): Either[T, T]
+  def touch(authenticator: T): Either[T, T]
```

Comment:
I had and have a lot of problem on my team about the understanding of left and right on Either primarily when use `Either` to return the same type because you need to read the code or the documentation to understand what is the left and right.
It's was only one suggestion, if you are ok with Either so keep going.